### PR TITLE
Add autocorrect/autocomplete

### LINKED ISSUE #51: Add Auto Complete for command

### PR DESCRIPTION
**Pull request by DaneMarc**
Closes #51 

Pressing `Ctrl` while typing a command will change the last typed word to the closest command word. Can both complete or correct a word.
E.g.
dele -> delete
feleteGiltereed-> deleteFiltered

### CODE REVIEW

File: src/main/java/seedu/address/ui/CommandBox.java [hunk 1]
```diff
@@ -100,6 +105,51 @@ private void setNextCommand() {
         commandTextField.end();
     }
 
+    private void autocomplete(String input) {
```
Comment (KeithCZW) [inline src/main/java/seedu/address/ui/CommandBox.java:108]: Should autocomplete be camelcase? Is it one word?

File: src/main/java/seedu/address/ui/CommandBox.java [hunk 2]
```diff
@@ -100,6 +105,51 @@ private void setNextCommand() {
...
+                    dp[i % 2][j] = i;
+                } else {
+                    dp[i % 2][j] = Math.min(dp[(i - 1) % 2][j - 1] + (str1.charAt(i - 1) == str2.charAt(j - 1) ? 0 : 1),
+                            Math.min(dp[i % 2][j - 1] + 1, dp[(i - 1) % 2][j] + 1));
+                }
+            }
+        }
+
+        return dp[m % 2][n];
+    }
```
Comment (KeithCZW) [inline src/main/java/seedu/address/ui/CommandBox.java:151]: This some cryptic stuff

**Review by KeithCZW — APPROVED** [review]
Other than the cryptic math stuff, LGTM!

File: src/main/java/seedu/address/ui/CommandBox.java (hunk 1 above)
Comment (DaneMarc) [reply src/main/java/seedu/address/ui/CommandBox.java:108]: it is to me 😌

**Pull request merged**